session/pingpong/invoice_tracker_test.go

```diff
@@ -538,43 +531,45 @@ func TestInvoiceTracker_receiveExchangeMessageOrTimeout(t *testing.T) {
 			fields: fields{
 				exchangeMessageWaitTimeout: time.Minute,
 				exchangeMessageChan:        make(chan crypto.ExchangeMessage),
-				lastInvoice: lastInvoice{
-					invoice: crypto.Invoice{AgreementTotal: 10, Hashlock: "0x441Da57A51e42DAB7Daf55909Af93A9b00eEF23C"},
+				accountantCaller:           &mockAccountantCaller{},
+				accountantPromiseStorage:   &mockAccountantPromiseStorage{},
+				peer:                       identity.FromAddress(addr3),
+				registryAddress:            mockRegistryAddress,
+				channelImplementation:      mockChannelImplementation,
+				accountantID:               identity.FromAddress(mockAccountantAddress),
+				invoicesSent: map[string]sentInvoice{
+					hex.EncodeToString(msg3.Promise.Hashlock): sentInvoice{
+						invoice: crypto.Invoice{
+							Hashlock: hex.EncodeToString(msg3.Promise.Hashlock),
+						},
+					},
 				},
-				accountantCaller:         &mockAccountantCaller{},
-				accountantPromiseStorage: &mockAccountantPromiseStorage{},
-				peer:                     identity.FromAddress(addr3),
-				registryAddress:          mockRegistryAddress,
-				channelImplementation:    mockChannelImplementation,
-				accountantID:             identity.FromAddress(mockAccountantAddress),
 			},
 			em: &msg3,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			it := &InvoiceTracker{
-				peer:                       tt.fields.peer,
-				exchangeMessageChan:        tt.fields.exchangeMessageChan,
-				exchangeMessageWaitTimeout: tt.fields.exchangeMessageWaitTimeout,
-				accountantFailureCount:     tt.fields.accountantFailureCount,
-				accountantPromiseStorage:   tt.fields.accountantPromiseStorage,
-				accountantID:               tt.fields.accountantID,
-				lastExchangeMessage:        tt.fields.lastExchangeMessage,
-				accountantCaller:           tt.fields.accountantCaller,
-				lastInvoice:                tt.fields.lastInvoice,
-				registryAddress:            tt.fields.registryAddress,
-				publisher:                  &mockPublisher{},
-				invoiceStorage:             NewProviderInvoiceStorage(NewInvoiceStorage(bolt)),
-				channelAddressCalculator:   NewChannelAddressCalculator(tt.fields.accountantID.Address, tt.fields.channelImplementation, tt.fields.registryAddress),
+			deps := InvoiceTrackerDeps{
```

Comment:
Check unit tests coverage report. Would be good to have higher coverage for payments related stuff :)

Reply:
it should really be covered in an e2e flow, see:

https://github.com/mysteriumnetwork/node/issues/1330